code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm

```diff
@@ -469,6 +469,42 @@
 
 	user_xeno.hive.banished_ckeys.Remove(banished_name)
 
+/datum/action/xeno_action/onclick/toggle_mute/use_ability(atom/target)
+	var/mob/living/carbon/xenomorph/queen/queen_user = owner
+	if(!queen_user.check_state())
+		return
+
+	var/choice = tgui_input_list(queen_user, "Pick a Xenomorph:", "Toggle the Hivemind Connection of a Xenomorph", queen_user.hive.totalXenos)
+
+	if(!choice)
+		return
+
+	var/mob/living/carbon/xenomorph/target_xeno
+
+	for(var/mob/living/carbon/xenomorph/xeno in queen_user.hive.totalXenos)
+		if(html_encode(xeno.name) == html_encode(choice))
+			target_xeno = xeno
+			break
+
+	var/reason = tgui_input_text(queen_user, "Why is [target_xeno] being excommunicated?")
+
+	if(!reason)
+		return
+
+	var/datum/weakref/target_ref = WEAKREF(target_xeno)
+
+	if(target_ref in queen_user.hive.muted_xenos)
```

Comment:
This should probably be above the reason input

Reply:
yep!

Reply:
thanks btw, completely missed that somehow